### PR TITLE
Ask search and category pages

### PR DESCRIPTION
**Pull request by virginiacc**
Displays and paginates results on Ask search and category pages

## Additions

- Spanish category pages
- Paginated results on category and search pages
- Side nav on English search page
- Option for round icon in `category-icon.html`
- Ability to rename `prev` and `next` buttons in `pagination.html` 

## Testing

- Check out branch
- Set up Ask
  ```
  export DEPLOY_ENVIRONMENT=build
  ./drop-db.sh
  ./create-mysql-db.sh
  mysql -uroot v1 < ~/Downloads/production_django.sql
  python cfgov/manage.py migrate
  python cfgov/manage.py runscript migrate_knowledgebase
  ```
- Run gulp
- Navigate to [English](http://localhost:8000/ask-cfpb/category-student-loans/) and [Spanish](http://localhost:8000/obtener-respuestas/categoria-pagar-la-universidad/) category pages and [English](http://localhost:8000/ask-cfpb/search/?q=credit) and [Spanish](http://localhost:8000/obtener-respuestas/buscar-es/?q=credito) search pages and check that question results are displayed and paginated

## Screenshots

screencapture-localhost-8000-ask-cfpb-category-auto-loans-1494511459852
screencapture-localhost-8000-ask-cfpb-search-1494511502334
screen shot 2017-05-11 at 7 05 44 am
screen shot 2017-05-11 at 7 06 12 am

## Todos

- There's currently a failing test that I still need to fix

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/unprocessed/css/pages/ask.less [hunk 1]
```diff
@@ -65,4 +62,58 @@
...
+    }
+}
+
+.ask-search-page {
+
+    .question_summary {
+        padding: unit((@grid_gutter-width / 2) / @base-font-size-px, em) 0;
+
+        .respond-to-min(@bp-med-min, {
+            padding:  unit(@grid_gutter-width / @base-font-size-px, em) 0 unit((@grid_gutter-width / 2) / @base-font-size-px, em) 0;
```
Comment (sebworks) [inline cfgov/unprocessed/css/pages/ask.less]: Minor: stack the values here.

File: cfgov/unprocessed/css/pages/ask.less [hunk 2]
```diff
@@ -3,44 +3,41 @@
...
+        padding: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em ) 0;
+        margin: unit( @grid_gutter-width / @base-font-size-px, em ) 0;
+        color: @gray;
+
+        .respond-to-min(@bp-sm-min, {
+            padding: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
+        });
+
+        .respond-to-min(@bp-med-min, {
+           width: 80%;
```
Comment (sebworks) [inline cfgov/unprocessed/css/pages/ask.less]: why are you setting this to `80%`?

File: cfgov/unprocessed/css/pages/ask.less [hunk 3]
```diff
@@ -65,4 +62,58 @@
...
+
+/* Results pages */
+
+.ask-category-page {
+    h1 .cf-icon {
+        color: @green;
+    }
+
+    .lead-paragraph {
+        max-width: 600px;
```
Comment (sebworks) [inline cfgov/unprocessed/css/pages/ask.less]: I'm not sure why this is necessary, but if we are using it repeatedly then it make make sense to make it a variable.

File: cfgov/jinja2/v1/ask-cfpb/landing-page.html [hunk 4]
```diff
@@ -16,8 +16,8 @@
         {{ render_block.render(block, loop.index) }}
     {%- endfor %}
 
-    <main class="content" id="main" role="main">
-        <div class="wrapper content_wrapper ask-landing-page">
+    <main class="content ask-landing-page ask-cfpb-page" id="main" role="main">
```
Comment (sebworks) [inline cfgov/jinja2/v1/ask-cfpb/landing-page.html]: Is this `BEM`? Should it be `ask-cfpb-page ask-cfpb-page__landing`?

**Comment by sebworks** [comment]
This scope is limited so merge when ready. 👍

File: cfgov/unprocessed/css/pages/ask.less (hunk 2 above)
Comment (virginiacc) [inline cfgov/unprocessed/css/pages/ask.less]: On larger screens, the tips/warnings become slightly narrower so they stand out from the surrounding text:

screen shot 2017-05-15 at 1 53 37 pm

File: cfgov/unprocessed/css/pages/ask.less (hunk 3 above)
Comment (virginiacc) [inline cfgov/unprocessed/css/pages/ask.less]: I think it's for readability -- I'll create a variable to make that clear.

File: cfgov/jinja2/v1/ask-cfpb/landing-page.html (hunk 4 above)
Comment (virginiacc) [inline cfgov/jinja2/v1/ask-cfpb/landing-page.html]: Good point -- it should be, & I'll update.

**Comment by virginiacc** [comment]
Thanks for the feedback, @sebworks! I'm going to merge now so this can be integrated with #2913, but I'll make the suggested changes in a subsequent PR.

**Pull request merged**